src/lib/Gen/hbs/HandlebarsGenerator.cpp

```diff
@@ -15,13 +15,73 @@
 #include "Builder.hpp"
 #include "MultiPageVisitor.hpp"
 #include "SinglePageVisitor.hpp"
+
+#include "lib/Lib/TagfileWriter.hpp"
+#include "lib/Support/RawOstream.hpp"
+
+#include <llvm/ADT/SmallString.h>
+#include <llvm/Support/FileSystem.h>
+#include <llvm/Support/Path.h>
+
 #include <mrdocs/Support/Path.hpp>
+
+#include <fstream>
 #include <sstream>
 
 namespace clang {
 namespace mrdocs {
 namespace hbs {
 
+namespace {
+
+/** Generate a Tagfile for cross-referencing. */
+Error
+generateTagfile(
+    HandlebarsCorpus const& hbsCorpus,
+    std::string_view tagFilePath,
+    std::string_view generatedFilesPath,
+    bool isSinglePage)
```

Comment:
One common trick in SOLID is to avoid `bool` such as `isWhatever` because they don't represent what they do via types. 

When there are many parameters with categorical options, the usual solution is to create an enum for that. 

But when there's a single parameter, the usual solution is to rename `generateTagFile` to `generateTagFileImpl` and define `generateSinglePageTagFile` and `generateMultiPageTagFile` as short inline functions that call `generateTagFileImpl` with true and false to make the calling code more legible.